patches/react-native-share-menu+6.0.0.patch

```diff
@@ -1,13 +1,94 @@
+@@ -34,6 +34,8 @@ public class ShareMenuReactView: NSObject {
+             return
+         }
+ 
++        ShareMenuReactView.detachViewDelegate()
++
+         if error != nil {
+             let exception = NSError(
+                 domain: Bundle.main.bundleIdentifier!,
 diff --git a/node_modules/react-native-share-menu/ios/ReactShareViewController.swift b/node_modules/react-native-share-menu/ios/ReactShareViewController.swift
-index f42bce6..ee36062 100644
+index f42bce6..ab00813 100644
 --- a/node_modules/react-native-share-menu/ios/ReactShareViewController.swift
 +++ b/node_modules/react-native-share-menu/ios/ReactShareViewController.swift
 @@ -13,7 +13,7 @@ class ReactShareViewController: ShareViewController, RCTBridgeDelegate, ReactSha
    func sourceURL(for bridge: RCTBridge!) -> URL! {
  #if DEBUG
      return RCTBundleURLProvider.sharedSettings()?
 -      .jsBundleURL(forBundleRoot: "index.share", fallbackResource: nil)
-+      .jsBundleURL(forBundleRoot: "index.share")
++      .jsBundleURL(forBundleRoot: "index")
  #else
      return Bundle.main.url(forResource: "main", withExtension: "jsbundle")
  #endif
+@@ -29,20 +29,7 @@ class ReactShareViewController: ShareViewController, RCTBridgeDelegate, ReactSha
+       initialProperties: nil
+     )
+ 
+-    rootView.backgroundColor = UIColor(red: 1, green: 1, blue: 1, alpha: 1)
+-    backgroundColorSetup: if let backgroundColorConfig = Bundle.main.infoDictionary?[REACT_SHARE_VIEW_BACKGROUND_COLOR_KEY] as? [String:Any] {
+-      if let transparent = backgroundColorConfig[COLOR_TRANSPARENT_KEY] as? Bool, transparent {
+-        rootView.backgroundColor = nil
+-        break backgroundColorSetup
+-      }
+-
+-      let red = backgroundColorConfig[COLOR_RED_KEY] as? Float ?? 1
+-      let green = backgroundColorConfig[COLOR_GREEN_KEY] as? Float ?? 1
+-      let blue = backgroundColorConfig[COLOR_BLUE_KEY] as? Float ?? 1
+-      let alpha = backgroundColorConfig[COLOR_ALPHA_KEY] as? Float ?? 1
+-
+-      rootView.backgroundColor = UIColor(red: CGFloat(red), green: CGFloat(green), blue: CGFloat(blue), alpha: CGFloat(alpha))
+-    }
++    RCTBootSplash.initWithStoryboard("MainInterface", rootView: rootView)
+ 
+     self.view = rootView
+ 
+diff --git a/node_modules/react-native-share-menu/ios/ShareViewController.swift b/node_modules/react-native-share-menu/ios/ShareViewController.swift
+index 12d8c92..0d84f19 100644
+--- a/node_modules/react-native-share-menu/ios/ShareViewController.swift
++++ b/node_modules/react-native-share-menu/ios/ShareViewController.swift
+@@ -13,6 +13,7 @@ import MobileCoreServices
+ import UIKit
+ import Social
+ import RNShareMenu
++import Firebase
+ 
+ class ShareViewController: SLComposeServiceViewController {
+   var hostAppId: String?
+@@ -22,6 +23,10 @@ class ShareViewController: SLComposeServiceViewController {
+   override func viewDidLoad() {
+     super.viewDidLoad()
+     
++    if FirebaseApp.app() == nil {
```

Comment:
Hmm, this is another one that seems difficult to upstream. That said, it does seem useful to have the error boundary working... but at the same time, the error boundary component itself was designed to make sense within the main app (where you'd reload it).

If we're going to include the error boundary's functionality in terms of crashlytics reporting, I think it makes sense to work that into the design and talk about how it should look in the share extension.

If we did want to go down this road, in terms of the library changes, I imagine what we'd need to do is create classes meant to be extended by custom classes in the project, instead of just directly adding the file? Like... most libraries.

Reply:
Yeah there are a couple approaches we could take. We could instruct people to "Copy items if needed" instead of deliberately not doing that. Or maybe we could provide a way to hook into these particular lifecycle methods via swift extension or something similar. Either way I think providing a way to do some native setup would be a big improvement.